riscv/tests/riscv_data/poseidon_gl_via_coprocessor/src/main.rs

```diff
@@ -1,53 +1,36 @@
 #![no_main]
 #![no_std]
 
-use powdr_riscv_runtime::hash::{poseidon_gl, poseidon_gl_unsafe};
+use powdr_riscv_runtime::{
+    goldilocks::{Goldilocks, PRIME},
+    hash::poseidon_gl,
+};
 
 #[no_mangle]
 fn main() {
-    let mut i: [u64; 12] = [0; 12];
-    let h = poseidon_gl(&mut i);
+    let mut i = [Goldilocks::new(0); 12];
+    let h = poseidon_gl(&mut i).map(u64::from);
     assert_eq!(h[0], 4330397376401421145);
     assert_eq!(h[1], 14124799381142128323);
     assert_eq!(h[2], 8742572140681234676);
     assert_eq!(h[3], 14345658006221440202);
 
-    let mut i: [u64; 12] = [1; 12];
-    let h = poseidon_gl(&mut i);
+    let mut i = [Goldilocks::new(1); 12];
+    let h = poseidon_gl(&mut i).map(u64::from);
     assert_eq!(h[0], 16428316519797902711);
     assert_eq!(h[1], 13351830238340666928);
     assert_eq!(h[2], 682362844289978626);
     assert_eq!(h[3], 12150588177266359240);
 
-    let minus_one = 0xffffffff00000001 - 1;
-    let mut i: [u64; 12] = [minus_one; 12];
-    let h = poseidon_gl(&mut i);
+    let minus_one = PRIME - 1;
+    let mut i = [Goldilocks::new(minus_one); 12];
+    let h = poseidon_gl(&mut i).map(u64::from);
     assert_eq!(h[0], 13691089994624172887);
     assert_eq!(h[1], 15662102337790434313);
     assert_eq!(h[2], 14940024623104903507);
     assert_eq!(h[3], 10772674582659927682);
 
-    let mut i: [u64; 12] = [
-        18446744069414584321,
-        18446744069414584321,
-        18446744069414584321,
-        18446744069414584321,
-        18446744069414584321,
-        18446744069414584321,
-        18446744069414584321,
-        18446744069414584321,
-        0,
-        0,
-        0,
-        0,
-    ];
-    let h = poseidon_gl_unsafe(&mut i);
-    assert_eq!(h[0], 4330397376401421145);
-    assert_eq!(h[1], 14124799381142128323);
-    assert_eq!(h[2], 8742572140681234676);
-    assert_eq!(h[3], 14345658006221440202);
```

Comment:
This test doesn't make sense. With the new `Goldilocks` type, the nonsenseness is explicit.

The idea is to rely on the wraparound behavior of a value >= PRIME inside the PoseidonGL machine to make this input be equivalent to all zeros.

It relies too much on implementation details. It should be UB to pass a value that doesn't fit into Goldilocks to PoseidonGL.